client/scripts/views/modals/onboarding_modal/choose_wallet.ts

```diff
@@ -0,0 +1,83 @@
+import 'modals/onboarding_modal/choose_wallet.scss';
+
+import { Form, FormGroup, Input } from 'construct-ui';
+import m from 'mithril';
+import app from 'state';
+
+import { ChainBase } from '../../../models';
+import { onboardingChooseWalletIcon } from '../../components/sidebar/icons';
+import { ChainBaseIcon } from '../../components/chain_icon';
+import OnboardingFooterActions from './footer_actions';
+
+interface IOnboardingChooseWalletAttr {
+  onBack: () => void;
+  onNext: () => void;
+  selected: ChainBase;
+  onSelect: (base: ChainBase) => void;
+}
+
+interface IOnboardingChooseWalletState {
+  search: string;
+}
+
+const ChooseWallet: m.Component<IOnboardingChooseWalletAttr, IOnboardingChooseWalletState> = {
+  oninit: (vnode) => {
+    vnode.state.search = '';
+  },
+  view: (vnode) => {
+    const getWalletItemForChainBase = (base: ChainBase) => m('div.item', {
+      class: vnode.attrs.selected === base ? 'selected' : '',
+      onclick: () => {
+        vnode.attrs.onSelect(base);
+      }
+    }, [
+      m(ChainBaseIcon, { chainbase: base, size: 40 }),
+      base
+    ]);
+
+    const matchBase = (base: ChainBase) => {
+      if (!base) return false;
+      return base.toLowerCase().includes(vnode.state.search.toLowerCase());
+    };
+
+    const allChains = app.config.chains.getAll();
+    const sortedChainBases = [ChainBase.CosmosSDK, ChainBase.Ethereum, ChainBase.NEAR, ChainBase.Substrate].filter((base) => allChains.find((chain) => chain.base === base));
+
+    const chainbase = app.chain?.meta?.chain?.base;
+    const items = app.chain ? (matchBase(chainbase) ? [ getWalletItemForChainBase(chainbase) ] : []) : sortedChainBases.filter((base) => matchBase(base)).map((base) => getWalletItemForChainBase(base));
```

Comment:
This should be much easier to fetch using the WebWallet controller.

Reply:
not sure how to make this simpler with webwallet, could you help me, please?

Reply:
I can use `WebWalletController` `availableWallets` to retrieve available wallets.. Also need to map to chainbases